bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoDimmerHandler.java

```diff
@@ -119,13 +114,48 @@ public void initialize() {
     public void dispose() {
         logger.debug("WeMoDimmerHandler disposed.");
 
-        ScheduledFuture<?> job = refreshJob;
+        ScheduledFuture<?> job = this.pollingJob;
         if (job != null && !job.isCancelled()) {
             job.cancel(true);
         }
-        refreshJob = null;
+        this.pollingJob = null;
 
         removeSubscription();
+        service.unregisterParticipant(this);
+    }
+
+    private void poll() {
+        synchronized (jobLock) {
+            if (pollingJob == null) {
+                return;
+            }
+            try {
+                logger.debug("Polling job");
+
+                // First check if the Wemo device is set in the UPnP service registry
+                // If not, set the thing state to OFFLINE and wait for the next poll
+                if (!isUpnpDeviceRegistered()) {
+                    logger.debug("UPnP device {} not yet registered", getUDN());
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "upnp device not registered [\"" + getUDN() + "\"]");
+                    synchronized (upnpLock) {
+                        subscriptionState = new HashMap<>();
+                    }
+                    return;
+                }
+                if (host.isEmpty()) {
+                    URL descriptorURL = service.getDescriptorURL(this);
+                    if (descriptorURL != null) {
+                        host = substringBetween(descriptorURL.toString(), "://", ":");
```

Comment:
No, quite difficult, as it relys on the UPnP service registry...

Reply:
No, I mean simply the logic for extracting host from descriptorURL. Even though it's just one line, it's still duplicated in all handlers. More over, the custom way of parsing the URL could probably be refined a bit, but right now it would need to be changed in each class. Actually, could it be written as `host = descriptorURL.getHost();` simply?

Reply:
Moved to WemoUtil.java.

Reply:
Sorry for coming up with two suggestions at once and sending mixed signals. :-) When looking at this again: After simplifying to `descriptorURL.getHost()` instead of custom parsing I actually don't think it's necessary to centralize it, since you are no longer using own logic (which could be unit tested, changed, etc.). Now using the standard URL parser nothing should need to change. You can leave it in the utility class if you'd like, I don't have any problems with that. But consider if you prefer to simply call `getHost() `directly from here.